src/dsp_tools/commands/xmlupload/models/deserialise/deserialise_value.py

```diff
@@ -90,6 +110,18 @@ def __init__(
             self.value = "".join(node.itertext())
         self.link_uuid = node.attrib.get("linkUUID")  # not all richtexts have a link, so this attribute is optional
 
+    @classmethod
+    def factory_for_knora_api_tags(
+        cls, value: str, comment: str | None, permissions: str | None, link_uuid: str | None
+    ) -> Self:
+        instance = cls.__new__(cls)
+        instance.value = value
+        instance.resrefs = None
+        instance.comment = comment
+        instance.permissions = permissions
+        instance.link_uuid = link_uuid
+        return instance
```

Comment:
Me too, I felt a bit uncomfortable while writing this code, but I really don't see an alternative. The problem is that I need to bypass the constructor, because I don't have a suitable `etree._Element`. So my only alternative would be to create an artificial `etree._Element`, which is even less elegant than what I have here.

Or, another alternative would be to rework the entire architecture of the classes `XMLResource`, `XMLProperty`, and `XMLValue`, but that's out of scope here.

To conclude: If we find an alternative, I'm more than happy to change this.

Reply:
oooff... yea, that is annoying.  
I haven't thought that all the way through, so maybe it's not even viable. But could you simply move all the logic from the constructor to a factory method and make the constructor accept the actual values (maybe it would even work with a dataclass)?

But if that doesn't work, feel free to leave it as is - as long as it works...

Reply:
Thanks, this actually works, and I like it much better! See 3e44cd1228da18dd995bd495889d96660c0a5d4e

Reply:
Great! I wonder why I didn't think of that earlier - these classes with their constructors have caused me plenty of pain...  
We should (by no means in this PR! that's for a follow-up) do the same thing for `XMLProperty` and `XMLResource`